sympy/physics/units/util.py

```diff
@@ -149,8 +149,16 @@ def quantity_simplify(expr):
                 coeff = coeff * arg.scale_factor
         expr = coeff
 
-    if any(isinstance(a, Add) for a in expr.args):
-        return expr.func(*map(quantity_simplify, expr.args))
+    if isinstance(expr, Mul) and any(isinstance(a, Add) for a in expr.args):
```

Comment:
The problem is not there, really. How about:

```diff
diff --git a/sympy/physics/units/util.py b/sympy/physics/units/util.py
index cc15b65..041a608 100644
--- a/sympy/physics/units/util.py
+++ b/sympy/physics/units/util.py
@@ -133,6 +133,7 @@ def get_total_scale_factor(expr):
 def quantity_simplify(expr):
     if expr.is_Atom:
         return expr
+
     if not expr.is_Mul:
         return expr.func(*map(quantity_simplify, expr.args))

@@ -149,6 +150,7 @@ def quantity_simplify(expr):
     coeff, args = expr.as_coeff_mul(Quantity)
     args_pow = [arg.as_base_exp() for arg in args]
     quantity_pow, other_pow = sift(args_pow, lambda x: isinstance(x[0], Quantity), binary=True)
+    coeff *= Mul.fromiter([Pow(b, e, evaluate=False) for b, e in other_pow])
     quantity_pow_by_dim = sift(quantity_pow, lambda x: x[0].dimension)
     # Just pick the first quantity:
     ref_quantities = [i[0][0] for i in quantity_pow_by_dim.values()]
@@ -157,4 +159,4 @@ def quantity_simplify(expr):
             (quantity*i.scale_factor/quantity.scale_factor)**p for i, p in v)
             if len(v) > 1 else v[0][0]**v[0][1]
         for quantity, (k, v) in zip(ref_quantities, quantity_pow_by_dim.items())]
-    return coeff*Mul.fromiter(other_pow)*Mul.fromiter(new_quantities)
+    return coeff*Mul.fromiter(new_quantities)
```

Reply:
Oh, you are right. I was seeing it the other way.